### DISCUSSION TITLE
WinFSP.Launcher GPG CommandLine Registry Value

### DISCUSSION THREAD

**Discussion by ariefcfa** [winfsp/winfsp#558] [General]
## Context

I have WinFSP.Launcher for `rclone mount` working perfectly. Then i use `gpg --decrypt` symmetric in the WinFSP.Launcher `CommandLine` but is failed. gpg not prompting asking for the password

here is my CommandLine registry value
```
mount %\/b:1 %2 --config="\*\*\*" --password-command "gpg -q -d "D:\rclone_test\sym.gpg"" --log-file="D:\rclone_test\log\\%\\b1.log" --log-level=ERROR --metadata --no-update-modtime --retries-sleep=10s --track-renames --update --cache-dir="D:\rclone_test\cache\\%\\b1" --vfs-cache-mode=full --vfs-cache-max-age=8d --vfs-cache-max-size=3G --vfs-fast-fingerprint -o VolumePrefix=%\/_1 -o FileSecurity="D:P(A;;FA;;;WD)"
```

## Question

Why gpg won't run like usual in WinFSP.Launcher? I test to run `gpg --decrypt` in cmd running as SYSTEM using `PsExec.exe` and gpg running as expected there. Please kindly help

Thanks,

**Comment by billziss-gh** [comment]
I do not know enough about `gpg --decrypt` to give you a good answer.

As you already know the Launcher normally runs the file system under the SYSTEM account. This means that any home directory files that gpg needs to run will be found in the SYSTEM profile. Additionally the Launcher can redirect stdio which may trip `gpg` (but I believe it only does so if you use the `Credentials` or `Stderr` registry key).

BTW, the Launcher also has the ability to launch a file system under the account of the person who starts the file system. See the `RunAs="."` registry setting here: https://github.com/winfsp/winfsp/wiki/WinFsp-Registry-Settings. You may find that this helps you work around the `gpg` problem you are experiencing.

**Reply by ariefcfa** [reply]
Thanks. `RunAs` `dot` registry value is working. Now gpg prompting asking for password, and WinFSP.Launcher successfully perform `rclone mount`. But, i have to do this under SYSTEM account, because elevated programs can't access files in mounted non-elevated `rclone mount` (and vice versa)

I kinda got it, maybe i better ask this in gpg forum, about why gpg won't run like usual under system account. Thank you @billziss-gh

**Comment by ariefcfa** [comment]
progress report, some workaround. 

first, prepare gpg-agent.exe running as SYSTEM account. i use this command:
```
psexec -nobanner -s -i gpg-connect-agent /bye
```
_*) another way maybe is using Task Scheduler or Windows Services_

now, this command worked:
```
psexec \\127.0.0.1 -nobanner -s -i gpg -q -d ***\***\sym.gpg
```
i also tested this in `WinFSP.Launcher` now there is passphrase prompt.
so glad it worked

Regards.
p.s. also posted this at https://forum.gnupg.org/t/help-running-gpg-under-system-account-via-winfsp-launcher-for-rclone-mount/4651
